Wikipedia/UI-V5/WMFHomeViewController.m

```diff
@@ -427,6 +435,9 @@ - (void)controller:(id<WMFHomeSectionController>)controller didSetItems:(NSArray
 
 - (void)controller:(id<WMFHomeSectionController>)controller didAppendItems:(NSArray*)items {
     NSInteger section = [self indexForSectionController:controller];
+    if (section == NSNotFound) {
```

Comment:
@bgerstle this can happen during modifying the sections when the fetcher is working, but we are also potentially adding, removing sections. Since sections are added using the performBatchupdates method (which runs on a queue sometime in the future), its possible to get a callback when it is configured, but not quite loaded into the collection view yet - as the collection view is still processing the update.

As far as documenting… like I mentioned previously, the plan is to solidify the architecture when working on the ticket to actually implement the business logic for the sections. So thats coming soon. patience…